frontend/src/metabase/admin/settings/components/ApiKeys/CreateApiKeyModal.tsx

```diff
@@ -0,0 +1,81 @@
+import { t } from "ttag";
+import { useCallback, useState } from "react";
+
+import { Text, Button, Group, Modal, Stack } from "metabase/ui";
+import {
+  Form,
+  FormErrorMessage,
+  FormProvider,
+  FormGroupWidget,
+  FormSubmitButton,
+  FormTextInput,
+} from "metabase/forms";
+import { ApiKeysApi } from "metabase/services";
+
+import { SecretKeyModal } from "./SecretKeyModal";
+
+export const CreateApiKeyModal = ({
+  onClose,
+  refreshList,
+}: {
+  onClose: () => void;
+  refreshList: () => void;
+}) => {
+  const [modal, setModal] = useState<"create" | "secretKey">("create");
+  const [secretKey, setSecretKey] = useState<string>("");
+
+  const handleSubmit = useCallback(
+    async vals => {
+      const response = await ApiKeysApi.create(vals);
+      setSecretKey(response.unmasked_key);
+      setModal("secretKey");
+      refreshList();
+    },
+    [refreshList],
+  );
+
+  if (modal === "secretKey") {
+    return <SecretKeyModal secretKey={secretKey} onClose={onClose} />;
+  }
+
+  if (modal === "create") {
+    return (
+      <Modal
+        size="30rem"
+        padding="xl"
+        opened
+        onClose={onClose}
+        title={t`Create a new API Key`}
+      >
+        <FormProvider initialValues={{}} onSubmit={handleSubmit}>
+          <Form>
+            <Stack spacing="md">
+              <FormTextInput
+                name="name"
+                label={t`Key name`}
+                size="sm"
+                required
+              />
+              <FormGroupWidget
+                name="group_id"
+                label={t`Select a group to inherit its permissions`}
```

Comment:
Ryan flagged this phrasing as potentially confusing (I don't think we should use 'inheritance' here, as this is more about membership).

Maybe (as Ryan suggested):

> Which group should this key belong to?

Or we could elaborate:

```suggestion
                label={t`Which group should this key belong to? The key will have the same permissions granted to that group.`}
```